### ISSUE TITLE
Incorrect warning for extra in editable install

### ISSUE THREAD

**Issue by PyroGenesis** [pypa/pipenv#6071]
### Issue description

I tried to install a local path dependency with an extra `model_worker`. `pipenv` gives me a warning that it could not find the extra `model-worker` (the underscore changed to a dash). This warning comes up for `pipenv install` and `pipenv lock`.

However, when I checked the dependencies manually, it did seem to have installed them.

### Expected result

Simple successful dependency resolution without any warnings

### Actual result

```ps
PS > pipenv lock
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
Success!
[    ] Locking...Warning: WARNING:pipenv.patched.pip._internal.resolution.resolvelib.candidates:fschat 0.2.35 does not provide the extra 'model-worker'
Locking [dev-packages] dependencies...
Updated Pipfile.lock (9059219e653cb511d1f83207044aad05c340413b74fc0bcb356cc6ef495c134a)!
```

[verbose_log.txt](https://github.com/pypa/pipenv/files/14091855/verbose_log.txt)

### Steps to replicate

1. Run `pipenv shell`
2. Clone FastChat `git clone https://github.com/lm-sys/FastChat.git repo_fastchat`
3. Install it with the `model_worker` extra: `pipenv install -e .\repo_fastchat[model_worker]`

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2023.11.17'`

Pipenv location: `'C:\\Users\\User1\\AppData\\Roaming\\Python\\Python310\\site-packages\\pipenv'`

Python location: `'C:\\Program Files\\Python310\\python.exe'`

OS Name: `'nt'`

User pip version: `'23.3.2'`

user Python installations found:

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.10.11',
 'os_name': 'nt',
 'platform_machine': 'AMD64',
 'platform_python_implementation': 'CPython',
 'platform_release': '10',
 'platform_system': 'Windows',
 'platform_version': '10.0.20348',
 'python_full_version': '3.10.11',
 'python_version': '3.10',
 'sys_platform': 'win32'}
```

System environment variables:

  - `ALLUSERSPROFILE`
  - `APPDATA`
  - `CHOCOLATEYINSTALL`
  - `CHOCOLATEYLASTPATHUPDATE`
  - `CLIENTNAME`
  - `COMMONPROGRAMFILES`
  - `COMMONPROGRAMFILES(X86)`
  - `COMMONPROGRAMW6432`
  - `COMPUTERNAME`
  - `COMSPEC`
  - `CUDA_PATH`
  - `CUDA_PATH_V12_1`
  - `DRIVERDATA`
  - `HOMEDRIVE`
  - `HOMEPATH`
  - `LOCALAPPDATA`
  - `LOGONSERVER`
  - `NUMBER_OF_PROCESSORS`
  - `OS`
  - `PATH`
  - `PATHEXT`
  - `PIPENV_ACTIVE`
  - `PIPENV_VENV_IN_PROJECT`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PIP_PYTHON_PATH`
  - `PROCESSOR_ARCHITECTURE`
  - `PROCESSOR_IDENTIFIER`
  - `PROCESSOR_LEVEL`
  - `PROCESSOR_REVISION`
  - `PROGRAMDATA`
  - `PROGRAMFILES`
  - `PROGRAMFILES(X86)`
  - `PROGRAMW6432`
  - `PROMPT`
  - `PSMODULEPATH`
  - `PUBLIC`
  - `PYTHONDONTWRITEBYTECODE`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`
  - `SESSIONNAME`
  - `SYSTEMDRIVE`
  - `SYSTEMROOT`
  - `TEMP`
  - `TMP`
  - `USERDOMAIN`
  - `USERDOMAIN_ROAMINGPROFILE`
  - `USERNAME`
  - `USERPROFILE`
  - `VIRTUAL_ENV`
  - `WINDIR`

Pipenv–specific environment variables:

 - `PIPENV_ACTIVE`: `1`
 - `PIPENV_VENV_IN_PROJECT`: `1`

Debug–specific environment variables:

  - `PATH`: `C:\Users\User1\Documents\Personable\fastchat-test\.venv\Scripts;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\libnvvp;C:\Program Files\Python310\Scripts\;C:\Program Files\Python310\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.1.1\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Users\User1\AppData\Local\Microsoft\WindowsApps;C:\Users\User1\AppData\Roaming\Python\Scripts;C:\Users\User1\AppData\Roaming\Python\Python310\Scripts;`
  - `VIRTUAL_ENV`: `C:\Users\User1\Documents\Personable\fastchat-test\.venv`

---------------------------

Contents of `Pipfile` ('C:\\Users\\User1\\Documents\\Personable\\fastchat-test\\Pipfile'):

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
fschat = {extras = ["model_worker"], file = "repo_fastchat", editable = true}

[dev-packages]

[requires]
python_version = "3.10"

```

Contents of `Pipfile.lock` ('C:\\Users\\User1\\Documents\\Personable\\fastchat-test\\Pipfile.lock'):
[Pipfile.lock.txt](https://github.com/pypa/pipenv/files/14091922/Pipfile.lock.txt)

</details>

**Comment by matteius** [comment]
I believe this may be fixed on `2024.3.0`

**Comment by PyroGenesis** [comment]
Can confirm, warning does not appear anymore with `2024.3.0`
